roles/tackle/templates/secret-keycloak-db.yml.j2

```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+data:
+  POSTGRES_DATABASE: {{ rhsso_db_name_b64 }} 
+  POSTGRES_EXTERNAL_ADDRESS: {{ rhsso_db_host_b64 }}
+  POSTGRES_HOST:
+  POSTGRES_PASSWORD: {{ rhsso_db_pass_b64 }}
+  POSTGRES_USERNAME: {{ rhsso_db_user_b64 }}
+  POSTGRES_SUPERUSER: {{ "true" | b64encode }}
+  SSLMODE: {{ "prefer" | b64encode }}
```

Comment:
Does prefer mean we attempt TLS and fall back to non-TLS on failure?

Reply:
Yes. I believe from the postgres documentation this is the default. From what I could see what it was doing running its own DB is using `""` as the value so the same.

This lands in this env var on the keycloak DB and mu understanding is that it is a client side setting:
```
    - name: DB_CONNECTION_PROPERTY_sslmode
      value: prefer
```

`require` has the same meaning as `verify-ca` when a CA is  passed in, which it seems they do and therefore does not work.

This is honestly not great, but not worse than what we have today, I don't believe upstream or down. If that's true I'd rather we file an issue to improve the pgsql deployment upstream and down. If there's a quick win to fixing it here I'm open to that, but also still filing the issue for upstream keycloak and pathfinder.

Reply:
@jwmatthews @dymurray do you have a strong feeling about how/when we handle this? 

Depending how the 'client' to pathfinder is configured it may be that there is no SSL, SSL is required and not verifying, or is doing it all already.

Reply:
Docs are here: https://www.postgresql.org/docs/current/libpq-ssl.html

Reply:
Cool , as long as fallback is guaranteed then we should be ok, I believe pathfinder on the client side is probably going unencrypted by default , or perhaps falls back as well.